OrbitCore/StringManager.cpp

```diff
@@ -13,7 +13,14 @@ bool StringManager::AddIfNotPresent(uint64_t key, std::string_view str) {
   return true;
 }
 
-std::optional<std::string> StringManager::Get(uint64_t key) {
+bool StringManager::AddOrReplace(uint64_t key, std::string_view str) {
+  absl::MutexLock lock{&mutex_};
+  bool added = !key_to_string_.contains(key);
+  key_to_string_.insert_or_assign(key, str);
```

Comment:
This returns a pair whose first element is `true` if the insertion took place, this should save you the `contains`.